docker/developers/s3/cdo_sound_library/hoc_song_meta/populate.rb

```diff
@@ -0,0 +1,102 @@
+require_relative '../../populator'
+
+class CdoSoundLibrary::HocSongMeta::Populate
+  include Populator
+
+  # The API path that maps to the bucket
+  API_PATH = '/api/v1/sound-library/hoc_song_meta'
+
+  TEST_MUSIC_URL = 'https://curriculum.code.org/media/uploads/synthesize.mp3'.freeze
+  TEST_MUSIC_BPM = 110.0
+
+  def ffmpeg?
+    unless defined?(@ffmpeg)
+      @ffmpeg = !!system('which ffmpeg > /dev/null')
+    end
+
+    @ffmpeg
+  end
+
+  def download_test_music
+    to = local_path("synthesize.mp3")
+    unless File.exist?(to)
+      url = 'https://curriculum.code.org/media/uploads/synthesize.mp3'
+      response = HTTParty.get(url)
+      if response.code != 200
+        puts "ERROR: Cannot find the given file"
+        return
+      end
+
+      # Write out file
+      relative_path = File.path(Pathname.new(to).relative_path_from(base_path))
+      puts "Writing test music: #{bucket_name}:#{relative_path}"
+      data = response.body
+      File.open(to, 'w+') do |f|
+        f.binmode
+        f.write(data)
+      end
+    end
+
+    to
+  end
+
+  # Hook into the populate call to produce the appropriate mp3 as well
+  def populate(path = nil)
+    data = super(path)
```

Comment:
Because we are overriding the call, basically. The call usually "downloads" the file, but we want to do something after we download the file... here, we want to parse it.

The `download` call is probably best seen as a private or protected call. It duplicates a file that we have upstream via our `studio.code.org/api` calls to kinda lazily populate our local buckets... gets around the need for AWS access. The `populate` call is there to be overridable and is intended to be called when a request for that file happens.